data/libs/Calibration.lua

```diff
@@ -0,0 +1,118 @@
+local Vmax = 3021 -- maximum free available cargo of largest ship
+local Vmin = 12 -- maximum free available cargo of smallest ship
+local RVab = (Vmax / Vmin)^(1/Nships)
+
+-- Ratio Rab of profits of 'a' and 'b' ships should be eye detectible by player. So, we want it above 1.20.
+-- As i wanted between 1.20 and 1.25, i chosed, in a try not to get too away from current price of precious metals, to define it as 1.225
+local Rab = 1.225
+
+-- The exponential of former line 48 of SpaceStation.lua calculated by two ratios of volume and profit
+local Exponential = function (RVolume, Rprofit)
+	return math.log(RVolume)/(math.log(RVolume)-math.log(Rprofit))
+end
+
+local minMissiles = 2 --number of missiles that corresponds to minmum trading profit. In other words number of missiles smalest ship can acquire.
+local maxMissiles = 24 --maximum number of missiles any ship can acquire. It should be get average reward about equal to the profit of the second in hierarchy trading ship.
+
+local Calibration
+Calibration = {
+	--StockPriceOne can variate to fine tune stock and prices. I prefer it constant so as to define prices from stock. See function updateEquipmentStock in SpaceStation.lua.
+	--On low prices rn are large integers that can not exceed much the 2*10^9 anyway.
+	StockPriceOne = 10^9,
+	PriceExponential = Exponential(RVab, Rab),
+	avgStock = v
+}
+
+-- THE MOST IMPORTANT FUNCTION OF CALIBRATION. Average maximum profit of a ship that can trade cargo tones of his specialization commoditiy.
+-- Once the initial and permanent settings for StockPriceOne and PriceExponential is taken, all future ships' profits and prices are defined according to them.
+Calibration.profit = function (cargo)
+	local a = Calibration.PriceExponential
+	local constant = Calibration.StockPriceOne
+	local coeff = (v2 * constant / cargo)^((1-a) / a)
+	local profitTo = p * v * coeff * constant
+	local profitFrom = p * v2 * coeff * constant / 2
+	return (profitTo + profitFrom) / 2
+end
+
+--Define the ratio of fighting rewards to minimum trading profit in linear way. Used in Assassination and Combat modules.
+Calibration.Fighting = function (missiles)
+	local ratio = Rab^(Nships - 1)
+	return math.max(1, (ratio - 1) / (maxMissiles - minMissiles) * missiles + 1 - 2 * (ratio - 1) / (maxMissiles - minMissiles))
+end
+
+-- The maximum profit of the base (smallest) ship.
+Calibration.minProfit = Calibration.profit(Vmin)
+
+--Nominal cash needed to trade all stock of a commodity depended on its nominal price.
+Calibration.InvestmentOfPrice = function (price)
+	return Calibration.StockPriceOne / price^(Calibration.PriceExponential - 1)
+end
+
+-- Adding prerequisite feature to ships in order to exclude them from function addRandomShipAdvert of SpaceStation.lua, if player is not their prequisite.
+-- Prerequisite is table for example prerequisites["Sinonatrix"] = {"Amphiesma", "Lunar Shuttle"}
+
+Calibration.prerequisite = function (pship,xship)
```

Comment:
This ship prerequisite feature is unlikely to ever be merged. Pioneer's ships do not follow a linear upgrade path, and in fact most ships in a ship class are "sidegrades" that have tradeoffs between cargo space, acceleration, weapons capability, effective range, etc. It's a neat idea for a mod, but it doesn't match any of the design decisions we've made so far.

Reply:
This feature has no place in lua. His right place is a table tag in ships json files. 
_With empty tables it exists but not used._
It is another matter if developers want ship designers to have the ability to create such chains of ships or not. 

The exponential can exist fine without that feature. On specific range of values for exponential and little a attention to ship prices, the trading system works fine without it. I do not know if the idea to have this "sidegrades" can me applied between chains of models of ships. The avatar design scope, in my mind, can be accomplished with chains of ships too. 

I actually had uploaded a mod that do not use that feature and played it for a while. The two reasons i prefer it are
a) i have a better feeling of progress, a game play related matter, and 
b) it was clear that i should skip some ships. 
However some person spent hours and thinking and efforts to design it, so as for me to be able to see it available in game. It does not seemed right to neglect his efforts because his ship does not fit in best strategy. The second reason is not game play related.

The other option can come from equilibrium equations of the game model, i describe in my article, but should let, on some degree,  the ship prices to evolve over time. That seemed too difficult to be accepted and i never tried it.

The feature affects mostly ship designers when they create ships that should never be picked by players.
_Having said that, the feature is not in the main scope of this PR_ and may omitted if you wish to. I